mne/preprocessing/tests/test_maxwell.py

```diff
@@ -0,0 +1,114 @@
+    sss_nonStd = Raw(sss_nonStd_fname, preload=True, proj=False,
+                     allow_maxshield=True)
+
+    # Create coils
+    all_coils, meg_info = maxwell._make_coils(raw.info)
+    picks = [raw.info['ch_names'].index(ch) for ch in [coil['chname']
+                                                       for coil in all_coils]]
+    coils = [all_coils[ci] for ci in picks]
+    ncoils = len(coils)
+
+    raw.pick_channels(all_coils[ci]['chname'] for ci in picks)
+    sss_std.pick_channels(all_coils[ci]['chname'] for ci in picks)
+    sss_nonStd.pick_channels(all_coils[ci]['chname'] for ci in picks)
+
+    int_order, ext_order = 8, 3
+    n_int_bases = int_order ** 2 + 2 * int_order
+    n_ext_bases = ext_order ** 2 + 2 * ext_order
+    nbases = n_int_bases + n_ext_bases
+
+    # Check number of bases computed correctly
+    assert_equal(maxwell.get_num_moments(int_order, ext_order), nbases)
+
+    # Check multipolar moment basis set
+    S_in, S_out = maxwell._sss_basis(origin=np.array([0, 0, 40]), coils=coils,
+                                     int_order=int_order, ext_order=ext_order)
+    assert_equal(S_in.shape, (ncoils, n_int_bases), 'S_in has incorrect shape')
+    assert_equal(S_out.shape, (ncoils, n_ext_bases),
+                 'S_out has incorrect shape')
+
+    # Check normalization
+    assert_allclose(np.linalg.norm(S_in, axis=0), np.ones((S_in.shape[1])),
+                    rtol=1e-12, err_msg='S_in normalization error')
+    assert_allclose(np.linalg.norm(S_out, axis=0), np.ones((S_out.shape[1])),
+                    rtol=1e-12, err_msg='S_out normalization error')
+
+    # Test sss computation at the standard head origin
+    raw_sss = maxwell.maxwell_filter(raw, origin=[0., 0., 40.],
+                                     int_order=int_order, ext_order=ext_order)
+
+    assert_array_almost_equal(raw_sss[:, :][0], sss_std[:, :][0], decimal=11,
+                              err_msg='Maxwell filtered data at standard '
+                              ' origin incorrect.')
+
+    # Confirm SNR is above 100
+    bench_rms = np.sqrt(np.mean(sss_std[:, :][0] ** 2, axis=1))
+    error = raw_sss[:, :][0] - sss_std[:, :][0]
+    error_rms = np.sqrt(np.mean(error ** 2, axis=1))
+    assert np.mean(bench_rms / error_rms) > 1000, 'SNR < 1000'
+
+    # Test sss computation at non-standard head origin
+    raw_sss = maxwell.maxwell_filter(raw, origin=[0., 20., 20.],
+                                     int_order=int_order, ext_order=ext_order)
+    assert_array_almost_equal(raw_sss[:, :][0], sss_nonStd[:, :][0],
+                              decimal=11, err_msg='Maxwell filtered data at '
+                              'non-std origin incorrect.')
+    # Confirm SNR is above 100
+    bench_rms = np.sqrt(np.mean(sss_nonStd[:, :][0] ** 2, axis=1))
+    error = raw_sss[:, :][0] - sss_nonStd[:, :][0]
+    error_rms = np.sqrt(np.mean(error ** 2, axis=1))
+    assert np.mean(bench_rms / error_rms) > 1000, 'SNR < 1000'
+
```

Comment:
Please add a test that saving and loading the result works, and that the data from the two matches.

Also, ensure that `_get_sss_rank` (https://github.com/mne-tools/mne-python/blob/master/mne/io/proc_history.py#L284) produces a correct result.

You don't have to do it in this PR, but note here that a subsequent PR should refactor the maxfilter-reading test (https://github.com/mne-tools/mne-python/blob/master/mne/io/tests/test_proc_history.py#L16) to be importable here so you can more fully test your info population. I think it will take a bit of work to get right.